Extension_CloudFlare_Plugin.php

```diff
@@ -245,75 +300,85 @@ private function fix_remote_addr() {
 	 * Modified by James Greene <james@cloudflare.com> to include IPV6 support
 	 * (original version only supported IPV4).
 	 * 21 May 2012
-	*/
-	/**
-	 * ipv4_in_range
-	 * This function takes 2 arguments, an IP address and a "range" in several
-	 * different formats.
-	 * Network ranges can be specified as:
-	 * 1. Wildcard format:     1.2.3.*
-	 * 2. CIDR format:         1.2.3/24  OR  1.2.3.4/255.255.255.0
-	 * 3. Start-End IP format: 1.2.3.0-1.2.3.255
-	 * The function will return true if the supplied IP is within the range.
-	 * Note little validation is done on the range inputs - it expects you to
-	 * use one of the above 3 formats.
+	 *
+	 * @param string $ip    The IPv4 address to check.
+	 * @param string $range The range to check against.
+	 *
+	 * @return bool True if the IP is within the range, false otherwise.
 	 */
 	private function ipv4_in_range( $ip, $range ) {
 		if ( strpos( $range, '/' ) !== false ) {
 			// $range is in IP/NETMASK format
 			list( $range, $netmask ) = explode( '/', $range, 2 );
 			if ( strpos( $netmask, '.' ) !== false ) {
 				// $netmask is a 255.255.0.0 format
-				$netmask = str_replace( '*', '0', $netmask );
+				$netmask     = str_replace( '*', '0', $netmask );
 				$netmask_dec = ip2long( $netmask );
-				return ( ip2long( $ip ) & $netmask_dec ) == ( ip2long( $range ) & $netmask_dec );
+
+				return ( ip2long( $ip ) & $netmask_dec ) === ( ip2long( $range ) & $netmask_dec );
 			} else {
-				// $netmask is a CIDR size block
-				// fix the range argument
-				$x = explode( '.', $range );
-				while ( count( $x )<4 ) $x[] = '0';
+				// $netmask is a CIDR size block fix the range argument.
+				$x     = explode( '.', $range );
+				$count = count( $x );  // Assign count to a variable.
+				while ( $count < 4 ) {
```

Comment:
This will cause an indefinite loop.  The `while` condition must check the `count` of `$x` for every iteration.  You can change it to `while ( true )` and `break` if the condition is met.

Reply:
After asking you about this you indicated that it would be okay as is. This should never result in an infinite loop